### ISSUE TITLE
GPG

### ISSUE THREAD

**Issue by macOneOone** [holzschu/a-shell#784]
Hello @holzschu ,

Hope you're doing fine.

Is it possible to have git with gpg support?

**Comment by holzschu** [comment]
I'll have a look, but that sounds difficult: both git and gpg are GNU-licensed products, so they cannot be embedded in an AppStore release, they would have to be compiled to WebAssembly, and WebAssembly is not that good with network-based interactions.

**Comment by NobodyXu** [comment]
For git, you can consider [gix](https://docs.rs/gix), it is a library written in Rust, with low level API.

I think you'd able to compile to to wasm without filesystem, network access or process spawning using low-level API.

**Comment by holzschu** [comment]
For git, we already have `lg2`, which is based on `libgit2`, and that works. But "git with gpg" only works with the original git, not with any of the clones (neither gix nor libgit2).

Projects written in Rust fail at cross-compiling more often than not (because they depend on crates that depend on crates and somewhere at the bottom there is a crate that assumes the word is either Unix or Windows). So far, there has been only one project written in Rust that I could cross-compile to WebAssembly, ripgrep, and even that one required editing the code in several places (out of dozens of attempts).

**Comment by NobodyXu** [comment]
> Projects written in Rust fail at cross-compiling more often than not (because they depend on crates that depend on crates and somewhere at the bottom there is a crate that assumes the word is either Unix or Windows).

Yeah many do have that problem, but IIRC gix just fixed a cross compilation error related to wasm, so gix does care about that.

They also have a few configurable features, which could disable features relying on supporting network/threading.

**Comment by NobodyXu** [comment]
> But "git with gpg" only works with the original git, not with any of the clones (neither gix nor libgit2)

Thanks, it's unfortunate that neither gix nor libgit2 support gpg.

**Comment by macOneOone** [comment]
> I'll have a look, but that sounds difficult: both git and gpg are GNU-licensed products, so they cannot be embedded in an AppStore release, they would have to be compiled to WebAssembly, and WebAssembly is not that good with network-based interactions.

Ok, thank you so much for the effort
